accepted/2021/net6.0-tfms/net6.0-tfms.md

```diff
@@ -0,0 +1,396 @@
+else applies.
+
+We don't want to model the Xamarin TFMs as a fallback but instead want them to
+apply **before** `netcoreapp` and `netstandard`. Usually the existence of the
+Xamarin TFM implementation is to bait a `netstandard` API and switch the
+platform specific implementation. Most of the Plugin type packages use this
+pattern and in fact the `netstandard` "implementation" is more like a reference
+assembly with just stubs that throw at runtime. `Xamarin.Essentials` is a good
+example of this. The same rational can be applied to `netcoreapp` where this
+would logically be the server implementation (or maybe a .NET Core 3.x desktop
+implementation).
+
+A `net6.0-maccatalyst` project referencing a NuGet package should behave as
+follows:
+
+* It should prefer `xamarin.ios` assets over `netcoreapp`, `netstandard`,
+  and `net5.0` assets.
+    - However, it should still prefer `net6.0` assets over `xamarin.ios`.
+    - It also shouldn't be compatible with any other existing Xamarin TFM (such
+      as `xamarin.mac`)
+* Generate NuGet warning [NU1701] when a `xamarin.ios` asset is being used
+    - Package 'packageId' was restored using 'xamarin.ios' instead the project
+      target framework 'net6.0-maccatalyst'. This package may not be fully
+      compatible with your project.
+* Should only use compatible `net5.0` based TFMs, namely `net5.0`, `net6.0`, and
+  `net6.0-maccatalyst`.
+    - Specifically, it should not accept `net6.0-ios`. The expectation is that
+      moving forward libraries that want to work on iOS and Mac Catalyst should
+      use `net6.0` or multi-target for `net6.0-ios` and `net6.0-maccatalyst`
+
+We want the same rules for `net6.0-ios` except that using `xamarin.ios` should
+not generate a warning. This reasoning results in these precedence rules:
+
+**net6.0-ios**
+
+1. `net6.0-ios`
+1. `net6.0`
+1. `xamarin.ios` (no warning)
+1. `net5.0`
+1. `netcoreapp3.1` – `1.0`
+1. `netstandard2.1` – `1.0`
+1. `net4.x` – `1.0` (warning)
+
+**net6.0-maccatalyst**
+
+1. `net6.0-ios`
+1. `net6.0`
+1. `xamarin.ios` ([NU1701] warning)
+1. `net5.0`
+1. `netcoreapp3.1` – `1.0`
+1. `netstandard2.1` – `1.0`
+1. `net4.x` – `1.0 ([NU1701] warning)
+
+**net6.0-macos**
+
+1. `net6.0-macos`
+1. `net6.0` - `5.0`
+1. `netcoreapp3.1` – `1.0`
+1. `netstandard2.1` – `1.0`
+1. `net4.x` – `1.0` ([NU1701] warning)
+
+**net6.0-tvos**
+
+1. `net6.0-tvos`
+1. `net6.0` - `5.0`
+1. `netcoreapp3.1` – `1.0`
+1. `netstandard2.1` – `1.0`
+1. `net4.x` – `1.0` ([NU1701] warning)
+
+## APIs
+
+We need to add platform detection APIs for Mac Catalyst:
+
+```C#
+namespace System
+{
+    public partial class OperatingSystem
+    {
+        public static bool IsMacCatalyst();
+        public static bool IsMacCatalystVersionAtLeast(int major, int minor = 0, int build = 0);
+    }
+}
+```
+
+## Platform support annotations
+
+It seems the Xamarin team wants to have a single assembly with bindings for the
```

Comment:
Can we clarify the purpose of having a single assembly?

Reply:
The point I was trying to make was less about a single assembly vs multiple but a having a combined set of bindings for Mac-based platforms. Is that not accurate?